lib/babl/schema/any_of.rb

```diff
@@ -0,0 +1,137 @@
+require 'values'
+require 'set'
+require 'babl/schema/static'
+require 'babl/schema/object'
+
+module Babl
+    module Schema
+        class AnyOf < Value.new(:choice_set)
+            attr_reader :choices
+
+            def initialize(choices)
+                flattened_choices = choices.flat_map { |doc| AnyOf === doc ? doc.choices : [doc] }.uniq
+                @choices = flattened_choices
+                super(flattened_choices.to_set)
+            end
+
+            def json
+                { anyOf: choices.map(&:json) }
+            end
+
+            # Perform simple transformations in order to reduce the size of the generated
+            # schema.
+            def simplify
+                simplify_single ||
```

Comment:
Each simplification method is responsible for returning the simplest schema, and they do that by recursively calling `simplify` on the result after applying their own transformations, if the result is of type `AnyOf`.